analysis_options.yaml

```diff
@@ -1,2 +1,215 @@
+    - curly_braces_in_flow_control_structures # personal preference
+    - directives_ordering
+    - do_not_use_environment # depends on project requirements
+    - empty_catches
+    - empty_constructor_bodies
+    - exhaustive_cases
+    - file_names
+#    - flutter_style_todos         # I find the required ":" annoying and pointless
+    - implementation_imports
+    - join_return_with_assignment
+    - leading_newlines_in_multiline_strings
+    - library_names
+    - library_prefixes
+    - lines_longer_than_80_chars
+# Too annoying when long strings with interpolations are broken into multiple lines which we do often
+#    - missing_whitespace_between_adjacent_strings
+    - no_default_cases             # experimental
+    - no_runtimeType_toString
+    - non_constant_identifier_names
+    - null_closures
+    - omit_local_variable_types
+# annoying when introducing a class to be used as interface that has only one member
+#    - one_member_abstracts
+    - only_throw_errors
+    - overridden_fields
+    - package_api_docs
+    - package_prefixed_library_names
+    - parameter_assignments        # personal preference
+    - prefer_adjacent_string_concatenation
+    - prefer_asserts_in_initializer_lists
+#    - prefer_asserts_with_message # personal preference
+#    - prefer_bool_in_asserts # obsolete in Dart 2
+    - prefer_collection_literals
+    - prefer_conditional_assignment
+    - prefer_const_constructors
+    - prefer_const_constructors_in_immutables
+    - prefer_const_declarations
+    - prefer_const_literals_to_create_immutables
+    - prefer_constructors_over_static_methods
+    - prefer_contains
+#    - prefer_double_quotes # I think Dart style guide prefers single quotes
+    - prefer_equal_for_default_values
+    - prefer_expression_function_bodies
+    - prefer_final_fields          # personal preference
+    - prefer_final_in_for_each     # personal preference
+    - prefer_final_locals          # personal preference
+    - prefer_for_elements_to_map_fromIterable
+    - prefer_foreach
+    - prefer_function_declarations_over_variables
+    - prefer_generic_function_type_aliases
+    - prefer_if_elements_to_conditional_expressions
+    - prefer_if_null_operators
+    - prefer_initializing_formals
+    - prefer_inlined_adds
+    - prefer_int_literals
+    - prefer_interpolation_to_compose_strings
+    - prefer_is_empty
+    - prefer_is_not_empty
+    - prefer_is_not_operator
+    - prefer_iterable_whereType
+    - prefer_mixin
+    - prefer_null_aware_operators
+    - prefer_single_quotes
+    - prefer_spread_collections
+    - prefer_typing_uninitialized_variables
+    - provide_deprecation_message
+#    - public_member_api_docs      # personal preference
+    - recursive_getters
+    - sized_box_for_whitespace
+    - slash_for_doc_comments
+    - sort_child_properties_last
+#      - sort_constructors_first           # personal preference
+#      - sort_unnamed_constructors_first   # personal preference
```

Comment:
We can enable these 2 settings: `sort_constructors_first` and `sort_unnamed_constructors_first`

Reply:
Ignoring this for now. We can track and decide if we'd need this later on.